kytos/core/interface.py

```diff
@@ -173,19 +177,295 @@ def status_reason(self):
             super().status_reason
         )
 
-    def set_available_tags(self, iterable):
+    @staticmethod
+    def range_intersection(
+        ranges_a: list[list[int]],
+        ranges_b: list[list[int]]
+    ) -> list[list[int]]:
+        """Returns the intersection between two list
+        of ranges"""
+        result = []
+        a_i, b_i = 0, 0
+
+        while a_i < len(ranges_a) and b_i < len(ranges_b):
+            fst_a, snd_a = ranges_a[a_i]
+            fst_b, snd_b = ranges_b[b_i]
+
+            # Moving forward with non-intersection
+            if snd_a < fst_b:
+                a_i += 1
+            elif snd_b < fst_a:
+                b_i += 1
+            else:
+                # Intersection
+                intersection_start = max(fst_a, fst_b)
+                intersection_end = min(snd_a, snd_b)
+                result.append([intersection_start, intersection_end])
+
+                if snd_a < snd_b:
+                    a_i += 1
+                else:
+                    b_i += 1
+        return result
+
+    def get_available_tags(self, tag_type: str = '1') -> list[list[int]]:
+        """Get available tags, if there is tag_ranges restriction
+        self.available_tags are not truly available"""
+        if self.tag_ranges[tag_type] is None:
+            return self.available_tags[tag_type]
+        return self.range_intersection(self.available_tags[tag_type],
+                                       self.tag_ranges[tag_type])
+
+    def verify_first_last_tag(
+        self,
+        tag_ranges: list[list[int, int]],
```

Comment:
Unfortunately the `list` type hint only allow a single arg. 

Looks like you could leverage `Annotated[list[int], 2]` if you want to make it clear, a create an alias for this new type. I haven't explored much, and also not sure if it would work out of the box with type linters in the future, https://docs.python.org/3/library/typing.html#typing.Annotated. The least surprising option would probably be to just keep `list[list[int]]` for ow, and if needed a bit of more clarify a few comments in the docstring

Reply:
Keeping `list[list[int]]` for now until there is another option.